File: 2831-number-of-beautiful-pairs/number-of-beautiful-pairs.cs
public class Solution {
    public int gcd(int a,int b){
        if(b==0) return a;
        return gcd(b,a%b);
    }
    public int CountBeautifulPairs(int[] nums) {
        int c=0;
        for(int i=0;i<nums.Length;i++){
            int first=nums[i].ToString()[0]-'0';
            for(int j=i+1;j<nums.Length;j++){
                if(gcd(first,nums[j]%10)==1) c++;
            }
        }
        return c;
    }
}

File: 2831-number-of-beautiful-pairs/number-of-beautiful-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countBeautifulPairs = function(nums) {
    function gcd(a,b){
        if(b==0) return a;
        return gcd(b,a%b);
    }
    let c=0;
    for(let i=0;i<nums.length;i++){
        let first=Number(nums[i].toString()[0]);
        for(let j=i+1;j<nums.length;j++){
            if(gcd(first,nums[j]%10)==1) c++;
        }
    }
    return c;
};